ssm_jax/bp/gauss_chain.py

```diff
@@ -0,0 +1,152 @@
+from functools import partial
+import chex
+import jax
+from jax import vmap, lax, jit
+from jax import numpy as jnp
+from ssm_jax.bp.gauss_bp import (
+    potential_from_conditional_linear_gaussian,
+    pair_cpot_condition,
+    pair_cpot_marginalise,
+    pair_cpot_absorb_message,
+    info_multiply,
+    info_divide
+)
+
+
+@chex.dataclass
+class GaussianChainPotentials:
```

Comment:
I vacillated on this and ended up removing it because the Bayesian in me was keen to keep the prior distinct but I agree that practically it is helpful to keep everything together.